/motion_capture_tracking:
  ros__parameters:
    type: "optitrack"
    hostname: "130.149.82.37"

    mode: "libobjecttracker" # one of motionCapture,libRigidBodyTracker

    topics:
      poses:
        qos:
          mode: "none"
          deadline: 100.0 # Hz

    logfilepath: './motion_capture_tracking/data/test' # use "", to disable

    marker_configurations:
      default: # for standard Crazyflie
        offset: [0.0, -0.01, -0.04]
        points:
          p0: [0.0177184,0.0139654,0.0557585]
          p1: [-0.0262914,0.0509139,0.0402475]
          p2: [-0.0328889,-0.02757,0.0390601]
          p3: [0.0431307,-0.0331216,0.0388839]
      default_single_marker:
        offset: [0.0, -0.01, -0.04]
        points:
          p0: [0.0177184,0.0139654,0.0557585]
      medium_frame:
        offset: [0.0, 0.0, -0.03]
        points:
          p0: [-0.00896228,-0.000716753,0.0716129]
          p1: [-0.0156318,0.0997402,0.0508162]
          p2: [0.0461693,-0.0881012,0.0380672]
          p3: [-0.0789959,-0.0269793,0.0461144]
      big_frame:
        offset: [0.0, 0.0, -0.06]
        points:
          p0: [0.0558163,-0.00196302,0.0945539]
          p1: [-0.0113941,0.00945842,0.0984811]
          p2: [-0.0306277,0.0514879,0.0520456]
          p3: [0.0535816,-0.0400775,0.0432799]

    dynamics_configurations:
      default:
        max_velocity: [2, 2, 3] # m/s
        max_angular_velocity: [20, 20, 10] # rad/s
        max_roll: 1.4 #rad
        max_pitch: 1.4 #rad
        max_fitness_score: 0.001

    rigid_bodies:
      crazyflie:
        initial_position: [0,0,0]
        marker: "default"
        # marker: "default_single_marker"
        dynamics: "default"